### PR TITLE
Add ability to include custom headers with requests

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Add custom headers from the local config as key value pairs

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12673

### CODE REVIEW

File: app/fetch_preconfig.js [hunk 1]
```diff
@@ -36,12 +36,27 @@ Client4.doFetchWithResponse = async (url, options) => {
         };
     }
 
+    const customHeaders = LocalConfig.CustomRequestHeaders;
+    let requestOptions = Client4.getOptions(options);
+    if (customHeaders &&
```
Comment (hmhealey) [inline app/fetch_preconfig.js]: Maybe have this just be `customHeaders && Object.keys(customHeaders)` to save space?

File: assets/base/config.json [hunk 2]
```diff
@@ -41,5 +41,7 @@
 
     "AppGroupId": "group.com.mattermost.rnbeta",
 
-    "ShowSentryDebugOptions": false
+    "ShowSentryDebugOptions": false,
+
+    "CustomRequestHeaders": null
```
Comment (hmhealey) [inline assets/base/config.json]: If we default this to `{}`, it might make it clear to whoever uses it that it can take any number of key/value pairs

**Comment by enahum** [comment]
@hmhealey changes made

File: app/fetch_preconfig.js [hunk 3]
```diff
@@ -36,12 +36,24 @@ Client4.doFetchWithResponse = async (url, options) => {
         };
     }
 
+    const customHeaders = LocalConfig.CustomRequestHeaders;
+    let requestOptions = Client4.getOptions(options);
+    if (customHeaders && Object.keys(customHeaders) > 0) {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/fetch_preconfig.js]:
```suggestion
    if (customHeaders && Object.keys(customHeaders).length > 0) {
```

**Comment by enahum** [comment]
Just used the new suggestion feature and it is amazing

**Comment by enahum** [comment]
@saturninoabril changes made

**Comment by saturninoabril** [comment]
That's cool.  I'll use it next time.

**Pull request merged**